crates/fluvio-hub-util/src/fvm/api/download.rs

```diff
@@ -32,26 +34,49 @@ async fn checksum(artf: &Artifact, path: &PathBuf) -> Result<()> {
     Ok(())
 }
 
+fn make_download_url(download_url: &str, params: Option<UrlQueryParams>) -> Result<Url> {
+    let mut url = Url::parse(download_url).map_err(|err| {
+        tracing::error!(%err, download_url=download_url, "Failed to parse Download URL into Uri instance");
+        Error::msg(err.to_string())
+    })?;
+
+    if let Some(url_query_params) = params {
+        url.set_query(Some(url_query_params.to_string().as_str()));
+    }
+
+    Ok(url)
+}
+
 #[async_trait]
 pub trait Download {
     /// Downloads the artifact to the specified directory
     ///
     /// Internally validates the checksum of the downloaded artifact
     /// and returns the path to the downloaded artifact
-    async fn download(&self, target_dir: PathBuf) -> Result<PathBuf>;
+    async fn download(
+        &self,
+        target_dir: PathBuf,
+        url_query_params: Option<UrlQueryParams>,
```

Comment:
I don't think we should add query_params to the artifact downloads.  These are served directly by public S3 buckets not the hub, and the service may not allow arbitrary query params.  We basically capture all we need from just the find pkgset event.

Reply:
Thats right! I will aim this to PkgSet endpoint instead.